_posts/2017-11-23-totw-109.md

```diff
@@ -0,0 +1,111 @@
+---
+title: "Tip of the Week #109: Meaningful `const` in Function Declarations"
+layout: tips
+sidenav: side-nav-tips.html
+published: false
+permalink: tips/109
+type: markdown
+order: "109"
+---
+
+Originally posted as totw/109 on 2016-01-14
+
+By Greg Miller (jgm@google.com)
+
+This document will explain when `const` is meaningful in function declarations,
+and when it is meaningless and best omitted. But first, let us briefly explain
+what is meant by the terms *declaration* and *definition*.
+
+Consider the following code.
+
+```c++
+void F(int);                     // 1: declaration of F(int)
+void F(const int);               // 2: re-declaration of F(int)
+void F(int) { /* ... */ }        // 3: definition of F(int)
+void F(const int) { /* ... */ }  // 4: error: re-definition of F(int)
+```
+
+The first two lines are function *declarations*. A function *declaration* tells
+the compiler the function's signature and return type. In the above example, the
+function's signature is `F(int)`. The constness of the function's parameter type
+is ignored, so both declarations are equivalent ([over.load][CppStandard]).
```

Comment:
Need precise link to the C++ standard, if possible, or unlinkify if not. (This only links to the base page and loses the over.load link.)

Also, my strong personal preference is to always name the heading being linked to. (In this case "Overloadable declarations".)

So I'd write this as.

See ["Overloadable declarations"](http://eel.is/c++draft/over.load) in the C++ standard.

Reply:
done